akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -228,6 +228,45 @@ class ConnectionPoolSpec extends AkkaSpec("""
       acceptIncomingConnection()
       val (Success(_), 42) = responseOut.expectNext()
     }
+
+    "never close hot connections when minConnections key is given and >0" in new TestSetup() {
```

Comment:
This would be rather a black box test than looking at the slot internals:

1. open connection pool with `minConnections = n`
2. `acceptIncomingConnection()` `n` times
3. send `n` requests with `Connection: close` at once
4. expect `n` responses
5. `acceptIncomingConnection()` `n` times